server/src/main/java/org/opensearch/search/aggregations/bucket/terms/IncludeExclude.java

```diff
@@ -79,6 +79,14 @@ public class IncludeExclude implements Writeable, ToXContentFragment {
     // can disagree on which terms hash to the required partition.
     private static final int HASH_PARTITIONING_SEED = 31;
 
+    /**
+     * The default length limit for a reg-ex string. The value is derived from {@link IndexSettings#MAX_REGEX_LENGTH_SETTING}.
+     * For context, see:
+     * https://github.com/opensearch-project/OpenSearch/issues/1992
+     * https://github.com/opensearch-project/OpenSearch/issues/2858
+     */
+    private static final int DEFAULT_MAX_REGEX_LENGTH = 1000;
```

Comment:
Is it possible for the [default value of the index setting](https://github.com/opensearch-project/OpenSearch/blob/908682d437ec744395030fe8c3f01973f596de20/server/src/main/java/org/opensearch/index/IndexSettings.java#L456) to use the same constant? I'm assuming if the default changed for one case it should probably change for the other as well?

Reply:
I did consider that. It would look like this:

```
private static final int DEFAULT_MAX_REGEX_LENGTH = IndexSettings.MAX_REGEX_LENGTH_SETTING.getDefault(Settings.EMPTY);
```

but since the goal of this change is to decouple `IndexSettings` from this class completely, I didn't want to tie the two together even through a constant. Wdyt? I could go either way :)

cc @rishabhmaurya

Reply:
A shared constant only couples them at the implementation level. You'd still be keeping IndexSettings out of the IncludeExclude public API, which is probably the most important part. I'll defer to your preference :)

Reply:
Ideally, it belongs to lucene `RegexExp` class, but I understand its not possible to patch lucene and incorporate this change in a short notice and that too for such a minor change. Maybe open a PR in lucene, so that we get it with next version. 

I believe all such constants, which can potentially be used across multiple modules, belong to a `opensearch.common` package. If you will see the usage of `IndexSettings` class across the codebase, it mostly because of the constants defined in it, which is a bad practice in my opinion. 
agree with [andrross](https://github.com/andrross) and will leave up to you.